lang/java/reef-common/src/main/java/org/apache/reef/client/parameters/DriverRestartHandlersConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.client.parameters;
+
+import org.apache.reef.driver.context.ActiveContext;
+import org.apache.reef.driver.parameters.DriverRestartCompletedHandlers;
+import org.apache.reef.driver.parameters.DriverRestartContextActiveHandlers;
+import org.apache.reef.driver.parameters.DriverRestartHandler;
+import org.apache.reef.driver.parameters.DriverRestartTaskRunningHandlers;
+import org.apache.reef.driver.task.RunningTask;
+import org.apache.reef.runtime.common.DriverRestartCompleted;
+import org.apache.reef.tang.formats.ConfigurationModule;
+import org.apache.reef.tang.formats.ConfigurationModuleBuilder;
+import org.apache.reef.tang.formats.OptionalImpl;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+public final class DriverRestartHandlersConfiguration extends ConfigurationModuleBuilder {
```

Comment:
@markusweimer Actually, on second thought, I will be renaming DriverRestartConfiguration to DriverRuntimeRestartConfiguration, since what will eventually be bound here are EventHandlers that depend on the runtime DriverRestartConfiguration (e.g. DriverRestartManager in YarnDriverRestartConfiguration). I will also be renaming DriverRestartHandlersConfiguration to DriverRestartConfiguration, make it Public and ClientSide.